docs/visualization.rst

```diff
@@ -0,0 +1,332 @@
+.. _plotting:
+
+======================
+Visualizing ND Objects
+======================
+
+.. _cube_plotting:
+
+Visualizing NDCubes
+===================
+`~ndcube.NDCube` provides a simple-to-use, yet powerful visualization method, `~ndcube.NDCube.plot`, which produces sensible visualizations based on the dimensionality of the data and optional user inputs.
+It is intended to be a useful quicklook tool and not a replacement for high quality plots or animations, e.g. for publications.
+Let's define an `~ndcube.NDCube` as with a shape of ``(4, 4, 5)`` and physical axes of helioprojective longitude, latitude and wavelength.  (Click the "Source code" link immediately below to see this `~ndcube.NDCube` instantiated.)
+
+.. plot::
+  :context: reset
+  :nofigs:
+
+  >>> import astropy.units as u
+  >>> import astropy.wcs
+  >>> import numpy as np
+  >>> from astropy.nddata import StdDevUncertainty
+
+  >>> from ndcube import NDCube
+
+  >>> # Define data array.
+  >>> data = np.random.rand(4, 4, 5)
+
+  >>> # Define WCS transformations in an astropy WCS object.
+  >>> wcs = astropy.wcs.WCS(naxis=3)
+  >>> wcs.wcs.ctype = 'WAVE', 'HPLT-TAN', 'HPLN-TAN'
+  >>> wcs.wcs.cunit = 'Angstrom', 'deg', 'deg'
+  >>> wcs.wcs.cdelt = 0.2, 0.5, 0.4
+  >>> wcs.wcs.crpix = 0, 2, 2
+  >>> wcs.wcs.crval = 10, 0.5, 1
+
+  >>> # Define mask.  Initially set all elements unmasked.
+  >>> mask = np.zeros_like(data, dtype=bool)
+  >>> mask[-1, -1][:] = True  # Now mask some values.
+  >>> # Define uncertainty, metadata and unit.
+  >>> uncertainty = StdDevUncertainty(np.sqrt(np.abs(data)))
+  >>> meta = {"Description": "This is example NDCube metadata."}
+  >>> unit = u.ct
+
+  >>> # Instantiate NDCube with supporting data.
+  >>> my_cube = NDCube(data, wcs=wcs, uncertainty=uncertainty, mask=mask, meta=meta, unit=unit)
+
+The plot method can be called very simply.
+
+.. plot::
+  :context:
+  :include-source:
+
+  >>> import matplotlib.pyplot as plt
+  >>> ax = my_cube.plot()
+  >>> plt.show()
+
+For a >2 array axes, as is the case above, an animation object is returned displaying either a line or image with sliders for each additional array axis.
+These sliders are used to sequentially update the line or image as it moves along its corresponding array axis, thus animating the data.
+By default and image animation is returned.
+(See below to learn how to use ``plot_axes`` kwarg to produce a line animation.)
+
+For for data with two array axes, an image is produced similar to that of `matplotlib.pyplot.imshow`.
+
+.. plot::
+  :context:
+  :include-source:
+
+  >>> ax = my_cube[0].plot()
+  >>> plt.show()
+
+Note that the bottom row of the image is empty because those array elements are marked ``True`` is ``.mask``.
```

Comment:
it isn't empty??

Reply:
No, I've defined the mask in the expanding code block above to demonstrate its affect.